File: jvm-client/src/test/java/com/tomaszezula/makker/client/jvm/java/GetBlueprint.java
package com.tomaszezula.makker.client.jvm.java;

import java.util.Arrays;

import static com.tomaszezula.makker.client.jvm.java.Config.makeClient;

public class GetBlueprint {
    public static void main(String[] args) {
        single();
        multiple();
    }

    private static void single() {
        makeClient.getBlueprint(471310)
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void multiple() {
        makeClient.getBlueprints(Arrays.asList(473703, 471310))
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }
}

File: jvm-client/src/test/kotlin/com/tomaszezula/makker/client/jvm/examples/GetBlueprint.kt
package com.tomaszezula.makker.client.jvm.examples

import io.kotest.common.runBlocking
import com.tomaszezula.makker.client.jvm.examples.Config.makeClient
import com.tomaszezula.makker.common.model.Scenario

fun main() {
    runBlocking {
        single()
        multiple()
    }
}

suspend fun single() {
    makeClient.getBlueprint(Scenario.Id(471310))
        .onSuccess {
            println(it)
        }.onFailure {
            it.printStackTrace()
        }
}

suspend fun multiple() {
    makeClient.getBlueprints(
        listOf(
            Scenario.Id(473703),
            Scenario.Id(471310),
        )
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}
